ietf-layer0-types-ext.yang

```diff
@@ -51,4 +51,397 @@ module ietf-layer0-types-ext {
+    description
+      "DP-QPSK (Dual Polarization Quadrature
+       Phase Shift Keying) modulation";
+  }
+
+  identity QAM8 {
+    base modulation;
+    description
+      "8QAM (8-State Quadrature Amplitude Modulation) modulation";
+  }
+
+  identity QAM16 {
+    base modulation;
+    description
+      "QAM16 (Quadrature Amplitude Modulation)";
+  }
+
+  identity DP-QAM8 {
+    base modulation;
+    description
+      "DP-QAM8 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+
+  identity DC-DP-QAM8 {
+    base modulation;
+    description
+    "DC DP-QAM8 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+
+  identity DP-QAM16 {
+    base modulation;
+    description
+      "DP-QAM16 (Dual Polarization Quadrature Amplitude Modulation)";
+  }
+
+  identity DC-DP-QAM16 {
+    base modulation;
+    description
+    "DC DP-QAM16 (Dual Polarization Quadrature 
+    Amplitude Modulation)";
+  } 
+
+  identity FEC {
+    description
+      "Base Identity the type of
+       Forward Error Correction";
+  }
+
+  identity reed-solomon {
+    base FEC;
+    description
+      "Reed-Solomon error correction";
+  }
+
+  identity hamming-code {
+    base FEC;
+    description
+      "Hamming Code error correction";
+  }
+
+  identity golay {
+    base FEC;
+      description "Golay error correction";
+  } 
+
+  identity line-coding {
+    description
+      "base line-coding class";
+    reference 
+      "ITU-T G.698.2-201811 section 7"   
+  }  
+
+  identity line-coding-NRZ-2p5G {
+    base line-coding;
+    description 
+      "ITU-T G.698.2-201811 section 7 table 8-1 "   
+  }
+
+  identity line-coding-NRZ-OTU1 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-2 "   
+  }
+
+  identity line-coding-NRZ-10G {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-3/8-5 "   
+  }
+
+  identity line-coding-NRZ-OTU2 {
+    base line-coding;
+    description
+      "ITU-T G.698.2-201811 section 7 table 8-4/8-6 "   
+
+  }
+
+// typedefs                
+
+  typedef operational-mode {
+    type string;
+    description
+      "Vendor-specific mode that guarantees 
+      interoperability.";
+    
+  }
+
+// temporary defined here for disalignment with 
+//ietf-layer0-types module
+  typedef standard-mode {
+    type string;
+    description
+      "ITU-T G.698.2 standard mode that guarantees 
+       interoperability.
+       It must be an string with the following format:
+       B-DScW-ytz(v) where all these attributes 
+       are conformant
+       to the ITU-T recomendation";
+    reference "ITU-T G.698.2 (11/2018)";
+  }
+
+  typedef frequency-thz {
+      type decimal64 {
+        fraction-digits 6;
+      }
+      units THz;
+      description
+        "The DWDM frequency in THz, e.g., 193.112500";
+      reference
+        "RFC6205: Generalized Labels for
+        Lambda-Switch-Capable (LSC) Label Switching Routers";
+  }
+  typedef frequency-ghz {

```

Comment:
this is a second definition for frequency and may lead to inconsistency... Why is this needed ?

Reply:
same comment as above. This is what was already proposed for layer0types .

Reply:
Sometimes (also in ITU-T) the Grid or n / m Granularity is specified in GigaHz while the central frequency is specified in THz.